docs/disaster-recovery.md

```diff
@@ -0,0 +1,443 @@
+
+```yaml=61
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplications", "volumereplicationclasses"]
+    verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplications/finalizers"]
+    verbs: ["update"]
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplications/status"]
+    verbs: ["get", "patch", "update"]
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplicationclasses/status"]
+    verbs: ["get"]
+```
+
+### Deploy the Volume Replication Sidecar
+
+Volume Replication Operator is a kubernetes operator that provides common
+ and reusable APIs for storage disaster recovery.
+ It is based on [csi-addons/spec](https://github.com/csi-addons/spec)
+ specification and can be used by any storage provider.
+
+To deploy `volume-replication` sidecar container in `csi-rbdplugin-provisioner`
+ pod, add the following yaml to
+ [csi-rbdplugin-provisioner deployment](https://github.com/ceph/ceph-csi/blob/release-v3.3/deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml).
+
+```yaml=1
+        - name: volume-replication
+          image: quay.io/csiaddons/volumereplication-operator:v0.1.0
+          args :
+            - "--metrics-bind-address=0"
+            - "--leader-election-namespace={{ .Namespace }}"
+            - "--driver-name=rbd.csi.ceph.com"
+            - "--csi-address=$(ADDRESS)"
+            - "--rpc-timeout=150s"
+            - "--health-probe-bind-address=:9998"
+            - "--leader-elect=true"
+          env:
+            - name: ADDRESS
+              value: unix:///csi/csi-provisioner.sock
+          imagePullPolicy: "IfNotPresent"
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /csi
+```
+
+## VolumeReplicationClass and VolumeReplication
+
+### VolumeReplicationClass
+
+*VolumeReplicationClass* is a cluster scoped resource that contains
+ driver related configuration parameters. It holds the storage admin
+ information required for the volume replication operator.
+
+### VolumeReplication
+
+*VolumeReplication* is a namespaced resource that contains references
+ to storage object to be replicated and VolumeReplicationClass
+ corresponding to the driver providing replication.
+
+>:bulb: For more information, please refer to the
+> [volume-replication-operator](https://github.com/csi-addons/volume-replication-operator).
+
+Let's say we have a *PVC* (rbd-pvc) in BOUND state; created using
+ *StorageClass* with `Retain` reclaimPolicy.
+
+```bash
+kubectl get pvc --context=cluster-1
+```
+
+>
+> ```bash
+> NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+> rbd-pvc   Bound    pvc-65dc0aac-5e15-4474-90f4-7a3532c621ec   1Gi        RWO            csi-rbd-sc   44s
+> ```
```

Comment:
combine these 2 to a single block

Reply:
I think it would be better if we keep the input and the output blocks separate just like all the other places. Also, all the output blocks have different formatting in output so that the user knows that they are just the output blocks, and nothing is needed to be done there. What do you think?

Reply:
AFAIK in cephcsi everything is in a single block in other files. we need to be consistant.